src/main/java/org/simlar/simlarserver/services/accountservice/AccountService.java

```diff
@@ -315,4 +319,14 @@ public void confirmAccount(final String simlarIdString, @SuppressWarnings("TypeM
     private static boolean checkRegistrationCodeFormat(final CharSequence input) {
         return input != null && REGEX_REGISTRATION_CODE.matcher(input).matches();
     }
+
+    public void deleteAccount(final SimlarId simlarId) {
+        if (simlarId == null) {
+            throw new IllegalArgumentException("simlarId=" + null);
+        }
```

Comment:
> If a caller passes null in some parameter for which null values are prohibited, convention dictates that
> NullPointerException be thrown rather than IllegalArgumentException.
Effective Java, 3rd Edition, Item 72.

```suggestion
        Objects.requireNonNull(simlarId, "simlarId=" + null);
```